packages/run-protocol/src/vaultFactory/addAssetToVault.js

```diff
@@ -0,0 +1,188 @@
+// @ts-check
+import { AmountMath, AssetKind } from '@agoric/ertp';
+import { makeRatio } from '@agoric/zoe/src/contractSupport/index.js';
+import { E } from '@endo/far';
+
+const reserveThenGetNames = async (nameAdmin, names) => {
+  for (const name of names) {
+    E(nameAdmin).reserve(name);
+  }
+  const nameHub = E(nameAdmin).readonly();
+  return Promise.all(names.map(name => E(nameHub).lookup(name)));
+};
+
+/**
+ * @param { EconomyBootstrapPowers } powers
+ * @param {*} config
+ */
+export const addInterchainAsset = async (
+  {
+    consume: { zoe, bankManager, agoricNamesAdmin, interchainMints },
+    produce: { interchainMints: produceInterchainMints },
+    installation: {
+      consume: { mintHolder },
+    },
+  },
+  {
+    options: {
+      denom,
+      keyword = 'IbcATOM',
+      proposedName = 'ATOM',
+      decimalPlaces = 4,
+    },
+  },
+) => {
+  /** @type {import('@agoric/vats/src/mintHolder.js').AssetTerms} */
+  const terms = {
+    keyword,
+    assetKind: AssetKind.NAT,
+    displayInfo: { decimalPlaces, assetKind: AssetKind.NAT },
+  };
+  const { creatorFacet: mint, publicFacet: issuer } = E.get(
+    E(zoe).startInstance(mintHolder, {}, terms),
+  );
+
+  const brand = await E(issuer).getBrand();
+  const kit = { mint, issuer, brand };
+
+  // Create the mint list if it doesn't exist and wasn't already rejected.
+  produceInterchainMints.resolve([]);
+  Promise.resolve(interchainMints).then(
+    mints => mints.push(mint),
+    () => {}, // If the interchainMints list was rejected, ignore the error.
+  );
```

Comment:
isn't `interchainMints` a promise?
and the error handling is already optional so no point in sending a noop.
```suggestion
  interchainMints.then(mints => mints.push(mint));
```

Reply:
Without the noop and if interchainMints is resolved to a rejection, this will produce a fresh unhandled rejection which we don't want.  Leaving as-is.

Reply:
> produce a fresh unhandled rejection which we don't want

To make sure I understand, this is to be explicit about ignoring the rejection? Is there a functional difference from noop handling or merely never handling? (in this case in which there's no await)